### PR TITLE
fix bug: close the rpc connection after use

### PR DESCRIPTION
**Pull request by Huster-ljw**
### What problem does this PR solve? 
every time the server apply for a connection when synchronizing with leader, but don't close it after finish it, which leads to run out of file descriptors. this PR solved the problem

### What is changed and how it works?
add close operation after finish it and the usage amount of file descriptor is normal

### CODE REVIEW

**Comment by nolouch** [comment]
Thanks, @Huster-ljw , please sign the cla.

File: server/region_syncer/client.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ func (s *RegionSyncer) establish(addr string) (ClientStream, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	defer cc.Close()
```
Comment (disksing) [inline server/region_syncer/client.go]: It closed, can the returned `ClientStream` be used?

Comment (Huster-ljw) [reply server/region_syncer/client.go]: cc is an instance of grpc.ClientConn, not ClientStream

Comment (Huster-ljw) [reply server/region_syncer/client.go]: there is something wrong. i have fixed it

File: server/region_syncer/client.go [hunk 2]
```diff
@@ -49,19 +49,19 @@ func (s *RegionSyncer) reset() {
 	s.regionSyncerCancel, s.regionSyncerCtx = nil, nil
 }
 
-func (s *RegionSyncer) establish(addr string) (ClientStream, error) {
+func (s *RegionSyncer) establish(addr string) (ClientStream, *grpc.ClientConn, error) {
```
Comment (shafreeck) [inline server/region_syncer/client.go]: How about wrap the` ClientStream` and `grpc.ClientConn` together?

**Comment by nolouch** [comment]
Thanks, @Huster-ljw, the ci still failed. can you check it with:
```
make dev
```
and you need to format your code with  `gofmt`
also, you need to sign the [cla](https://cla-assistant.io/pingcap/pd?pullRequest=1989).

**Comment by nolouch** [comment]
@Huster-ljw  Plase sign the cla.

**Pull request merged**